src/Interpolations.jl

```diff
@@ -123,131 +123,126 @@ function copy_with_padding(A, it::InterpolationType)
     coefs, pad
 end
 
-# This creates getindex methods for all supported combinations
-for IT in (
-        Constant{OnGrid},
-        Constant{OnCell},
-        Linear{OnGrid},
-        Linear{OnCell},
-        Quadratic{Flat,OnCell},
-        Quadratic{Flat,OnGrid},
-        Quadratic{Reflect,OnCell},
-        Quadratic{Reflect,OnGrid},
-        Quadratic{Line,OnGrid},
-        Quadratic{Line,OnCell},
-        Quadratic{Free,OnGrid},
-        Quadratic{Free,OnCell},
-        Quadratic{Periodic,OnGrid},
-        Quadratic{Periodic,OnCell},
-    )
-    for EB in (
-            ExtrapError,
-            ExtrapNaN,
-            ExtrapConstant,
-            ExtrapLinear,
-            ExtrapReflect,
-            ExtrapPeriodic,
-        )
-        it = IT()
-        eb = EB()
-        gr = gridrepresentation(it)
-
-        function getindex_impl(N)
-                quote
-                # Handle extrapolation, by either throwing an error,
-                # returning a value, or shifting x to somewhere inside
-                # the domain
-                $(extrap_transform_x(gr,eb,N))
-
-                # Calculate the indices of all coefficients that will be used
-                # and define fx = x - xi in each dimension
-                $(define_indices(it, N))
-
-                # Calculate coefficient weights based on fx
-                $(coefficients(it, N))
-
-                # Generate the indexing expression
-                @inbounds ret = $(index_gen(degree(it), N))
-                ret
-            end
-        end
+supported(::Type{Constant{OnGrid}}) = true
```

Comment:
No uses yet. I was keying off the `# This creates getindex methods for all supported combinations` and wondered if you wanted to be able to return an error like `"not a supported combination"` from something. So since the list was handy, I turned it into a trait. If we don't end up using it at all, then these should be deleted.